File: frontend/src/app/services/audit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Audit {
  id: string;
  action: string;
  timestamp: string;
  target_type: string;
  target_id: string;
  target_label?: string;
  user_id: string;
  username?: string;
  group_id?: string;
  old_values?: any;
  new_values?: any;
}

export interface Group {
  id: string;
  name: string;
  description?: string;
}



@Injectable({
  providedIn: 'root'
})
export class AuditService {

  private apiUrl = '/api/audits';

  constructor(private http: HttpClient) {}

  getGroups(): Observable<Group[]> {
    return this.http.get<Group[]>('/api/groups/');
  }

  getGroupAudits(groupId: string): Observable<Audit[]> {
    return this.http.get<Audit[]>(`${this.apiUrl}/groups/${groupId}`);
  }


}

File: backend/services/audit_service.py
from datetime import datetime
from backend.database import db


def create_audit(
    action: str,
    target_type: str,
    current_user: str,
    group_id: str,
    target_id: str,
    target_label: str = None,
    old_values: dict = None,
    new_values: dict = None
):
    db.audits.insert_one({
        "action": action,
        "timestamp": datetime.utcnow(),
        "target_type": target_type,
        "target_id": target_id,
        "target_label": target_label,
        "user_id": current_user,
        "group_id": group_id,
        "old_values": old_values,
        "new_values": new_values
    })